syntax = "proto3";

package catalogue;

service CatalogueService {
  rpc GetAllItems(Empty) returns (ItemList);
  rpc SearchItems(SearchRequest) returns (ItemList);
  rpc CreateItem(CreateItemRequest) returns (ItemResponse);
  rpc CreateSeller(CreateSellerRequest) returns (SellerResponse);
  rpc GetAllSellers(Empty) returns (SellerList);
}

message Empty {}

message SearchRequest {
  string keyword = 1;
}

message CreateItemRequest {
  string title = 1;
  string description = 2;
  double starting_price = 3;
  int32 duration_hours = 4;
  int32 seller_id = 5;
}

message ItemResponse {
  int32 id = 1;
  string title = 2;
  string description = 3;
  double starting_price = 4;
  double current_price = 5;
  string auction_type = 6;
  bool active = 7;
  int32 duration_hours = 8;
  string created_at = 9;
  string end_time = 10;
  int32 seller_id = 11;
  int32 remaining_time_seconds = 12;
}

message ItemList {
  repeated ItemResponse items = 1;
}

message CreateSellerRequest {
  string name = 1;
  string email = 2;
}

message SellerResponse {
  int32 id = 1;
  string name = 2;
  string email = 3;
}

message SellerList {
  repeated SellerResponse sellers = 1;
}
